.azure-pipelines/common.yml

```diff
@@ -0,0 +1,115 @@
+parameters:
+  name: ''
+  platform: ''
+  settings: ''
+  py_3_5: false
+  py_3_6: false
+  py_3_7: true
+  py_3_8: false
+  container_name: ''
+  container_startup: ''
+  additional_apt_gets: ''
+  additional_requirements: ''
+
+jobs:
+- job: ${{ format('{0}_{1}', parameters.name, parameters.platform) }}
+
+  timeoutInMinutes: 360
+
+  pool:
+    ${{ if eq(parameters.platform, 'Windows') }}:
+      vmImage: VS2017-Win2016
+    ${{ if eq(parameters.platform, 'Linux') }}:
+      vmImage: Ubuntu 16.04
+
+  strategy:
+    matrix:
+      ${{ if eq(parameters.py_3_5, 'true') }}:
+        Python35:
+          PYTHON_VERSION: '3.5'
+      ${{ if eq(parameters.py_3_6, 'true') }}:
+        Python36:
+          PYTHON_VERSION: '3.6'
+      ${{ if eq(parameters.py_3_7, 'true') }}:
+        Python37:
+          PYTHON_VERSION: '3.7'
+      ${{ if eq(parameters.py_3_8, 'true') }}:
+        Python38:
+          PYTHON_VERSION: '>= 3.8.0a'
+
+  steps:
+    - ${{ if ne(parameters.container_startup, '') }}:
+      - bash: ${{ parameters.container_startup }}
+        displayName: Spin up container
+
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(PYTHON_VERSION)'
+        architecture: 'x64'
+      displayName: Use Python version $(PYTHON_VERSION)
+
+    - ${{ if eq(parameters.platform, 'Linux') }}:
+      - bash: |
+          sudo apt-get -yq update
+          sudo apt-get -yq install libmemcached-dev
+          if [ -n "$OTHER_APT" ]; then
+            sudo apt-get -yq install $OTHER_APT
+          fi
+        displayName: Setup Linux environment
+        env:
+          OTHER_APT: ${{ parameters.additional_apt_gets }}
+
+    - bash: |
+        python -m pip install --upgrade pip setuptools wheel
+        pip install unittest-xml-reporting
+      displayName: Setup Python environment
+
+    - bash: |
+        pip install -r requirements/py3.txt

```

Comment:
We do install quite a lot; how about build caches to not poll from pypi all the time?

Reply:
We're working on our package caching story right now. The plan is to make this transparent, or at least trivial to opt into, so the line of script won't have to change.